nebula/ngStorage/StorageClient.py

```diff
@@ -200,15 +204,22 @@ def connect(self, address):
         Returns:
             - client: a storage client object
         """
+        retry = self._connection_retry
         if address not in self._clients.keys():
-            client = self.do_connect(address)
-            self._clients[address] = client
-            return client
+            while retry > 0:
+                client = self.do_connect(address)
+                if client != None:
+                    self._clients[address] = client
+                    return client
+                retry -= 1
         else:
             return self._clients[address]
 
     def disconnect(self, address):
```

Comment:
if the client is `None`, meaning the client is not added into the `_clients`, of course not need to call disconnect. Pay attention to the logic.